include/oneapi/dpl/pstl/iterator_impl.h

```diff
@@ -105,7 +106,13 @@ class zip_forward_iterator
     zip_forward_iterator() : __my_it_() {}
     explicit zip_forward_iterator(_Types... __args) : __my_it_(__tuple_t<_Types...>{__args...}) {}
 
-    reference operator*() const
+    reference
+    operator*() const
+#if _ONEDPL_CPP20_CONCEPTS_PRESENT
```

Comment:
What about instead of a adding a preprocessor guard and adding the `requires` expression inside, we introduce a macro function that checks `_ONEDPL_CPP20_CONCEPTS_PRESENT` internally and conditionally adds the constraint? I believe oneTBB does something like this.

Something like: `_ONEDPL_REQUIRES(std::indirectory_readable<_Types> &&...)` where the macro is defined as:
```
#if _ONEDPL_CPP20_CONCEPTS_PRESENT
#   define _ONEDPL_REQUIRES(req) \
    requires req
#else
#   define _ONEDPL_REQUIRES(req)
#endif
```

Reply:
Yeah, this is a good idea.  Thanks, I'll make this change.